widgetssdk/src/main/java/com/glia/widgets/core/chathead/domain/ResolveChatHeadNavigationUseCase.java

```diff
@@ -40,4 +44,8 @@ private boolean isMediaQueueingOngoing() {
     private boolean isMediaEngagementOngoing() {
         return engagementRepository.hasOngoingEngagement() && gliaEngagementTypeRepository.isMediaEngagement();
     }
+
+    private boolean isSharingScreen() {
+        return Dependencies.getUseCaseFactory().createIsCallVisualizerUseCase().execute();
```

Comment:
please turn  the `IsCallVisualizerUseCase` to constructor parameter

Reply:
I was on the fence if this should be the case, because the other constructor arguments were repositories.

Reply:
Just in case for the future, That doesn’t matter. 
We're passing instances to the constructor not to group instances, but to implement _`Dependency Injection`_